x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/stats/MemoryUsage.java

```diff
@@ -105,6 +119,7 @@ public String toString() {
     }
 
     public String documentId(String jobId) {
+        assert timestamp != null;
```

Comment:
What prevents `timestamp` from being `null` here?

Reply:
We should only persist a `MemoryUsage` object we read from `AnalyticsResultProcessor` and those should always have a non-null timestamp.